spring-cloud-task-jdbchdfs-common/src/main/java/org/springframework/cloud/task/jdbchdfs/common/JdbcHdfsConfiguration.java

```diff
@@ -0,0 +1,235 @@
+
+	@Bean
+	public JobExplorerFactoryBean jobExplorer() {
+		JobExplorerFactoryBean jobExplorerFactoryBean = new JobExplorerFactoryBean();
+		jobExplorerFactoryBean.setDataSource(this.dataSource);
+		return jobExplorerFactoryBean;
+	}
+
+	@Bean
+	public DeployerPartitionHandler partitionHandler(TaskLauncher taskLauncher, JobExplorer jobExplorer) throws Exception {
+		MavenResource resource = MavenResource.parse(taskResource);
+
+		DeployerPartitionHandler partitionHandler = new DeployerPartitionHandler(taskLauncher, jobExplorer, resource, "workerStep");
+
+		Map<String, String> environmentProperties = new HashMap<>();
+		environmentProperties.put("spring.profiles.active", "worker");
+
+		partitionHandler.setEnvironmentProperties(environmentProperties);
+		partitionHandler.setMaxWorkers(2);
+
+		return partitionHandler;
+	}
+
+	@Bean
+	@StepScope
+	public IncrementalColumnRangePartitioner partitioner(JobExplorer jobExplorer, @Value("#{stepExecutionContext['overrideCheckColumnValue']}") Long overrideValue) {
+		IncrementalColumnRangePartitioner partitioner = new IncrementalColumnRangePartitioner();
+		partitioner.setTable(props.getTableName());
+		partitioner.setColumn(props.getPartitionColumn());
+		partitioner.setJobExplorer(jobExplorer);
+		partitioner.setCheckColumn(props.getCheckColumn());
+		partitioner.setOverrideValue(overrideValue);
+		partitioner.setDataSource(dataSource);
+		partitioner.setPartitions(props.getPartitions());
+		return partitioner;
+	}
+
+	@Bean
+	@Profile("worker")
+	public DeployerStepExecutionHandler stepExecutionHandler(JobExplorer jobExplorer) {
+		DeployerStepExecutionHandler handler = new DeployerStepExecutionHandler(this.context, jobExplorer, this.jobRepository);
+
+		return handler;
+	}
+
+	@Bean
+	@StepScope
+	public HdfsTextItemWriter writer(@Value("#{stepExecutionContext['partSuffix']}") String suffix) throws Exception {
+
+		if (StringUtils.hasText(props.getFsUri())) {
+			hadoopConfiguration.set("fs.defaultFS", props.getFsUri());
+		}
+		if (StringUtils.hasText(props.getRmManagerPrinciple())) {
+			hadoopConfiguration.set("rm-manager-principal", props.getRmManagerPrinciple());
+		}
+		if (StringUtils.hasText(props.getNameNodePrinciple())) {
+			hadoopConfiguration.set("namenode-principal", props.getNameNodePrinciple());
+		}
+		if (StringUtils.hasText(props.getUserPrinciple())) {
+			hadoopConfiguration.set("user-principal", props.getUserPrinciple());
+		}
+		if (StringUtils.hasText(props.getUserKeyTab())) {
+			hadoopConfiguration.set("user-keytab", props.getUserKeyTab());
+		}
+		if (StringUtils.hasText(props.getSecurityMethod())) {
+			hadoopConfiguration.set("security-method", props.getSecurityMethod());
+		}
+		if (StringUtils.hasText(props.getPropertiesLocation())) {
+			hadoopConfiguration.set("properties-location", props.getPropertiesLocation());
+		}
+		if (StringUtils.hasText(props.getRegisterUrlHandler())) {
+			hadoopConfiguration.set("register-url-handler", props.getRegisterUrlHandler());
+		}
```

Comment:
Are these keys defined as constants anywhere?  Should we create them as constants?